foxai/explainer/computer_vision/object_detection/models/yolov5/yolo_object_detector.py

```diff
@@ -0,0 +1,279 @@
+"""File contains YOLOv5 ObjectDetector class."""
+import time
+from typing import List, Optional, Tuple
+
+import numpy as np
+import torch
+import torchvision
+from torchvision.ops.boxes import box_convert, box_iou
+
+from foxai.explainer.computer_vision.object_detection.base_object_detector import (
+    BaseObjectDetector,
+)
+from foxai.explainer.computer_vision.object_detection.models.yolov5.model import (
+    WrapperYOLOv5ObjectDetectionModel,
+)
+from foxai.explainer.computer_vision.object_detection.types import PredictionOutput
+
+
+class YOLOv5ObjectDetector(BaseObjectDetector):
+    """Custom YOLOv5 ObjectDetector class which returns predictions with logits to explain.
+
+    Based on code: https://github.com/pooya-mohammadi/yolov5-gradcam.
+    """
+
+    def __init__(
+        self,
+        model: WrapperYOLOv5ObjectDetectionModel,
+        img_size: Tuple[int, int],
+        names: List[str],
+        mode: str = "eval",
+        confidence: float = 0.4,
+        iou_thresh: float = 0.45,
+        agnostic_nms: bool = False,
+    ):
+        super().__init__()
+        self.device = model.device
```

Comment:
I am thinking that we should have it standardized across whole repository how we name ``device`` variable. Also how we get it to avoid possible confusions. because we either get it  
```python
device = model.device
```
or:
```python
    cuda = next(model.parameters()).is_cuda
    device = torch.device("cuda" if cuda else "cpu")
```

Reply:
I can't find the second way of obtaining the device in this branch.